keps/sig-node/20190717-seccomp-ga.md

```diff
@@ -193,31 +220,59 @@ for new profile sources to be added in the future (e.g. Kubernetes predefined pr
 profiles). The seccomp options struct leaves room for future extensions, such as defining the
 behavior when a profile cannot be set.
 
-<<[UNRESOLVED]>>
-What to do with the localhost profile type, given that we want to deprecate it? Alternative for
-consideration:
+##### LocalhostProfile
+
+This KEP proposes we GA LocalhostProfile as the only source of user-defined profiles at this point. 
+User-defined profiles are essential for users to realize the full benefits out of seccomp, allowing 
+them to decrease their attack surface based on their own workloads. 
+
+###### Updating seccomp profiles
```

Comment:
This doesn't feel like it's addressing the problem sufficiently to be called GA, but also implementing something more complex seems outside the scope for this KEP. This is a bit of chicken-and-egg situation, so maybe it's sufficient to just document the current behavior, and follow up with a separate feature (staged rollout) of a more complex solution.

For a proper solution, I was thinking of something along the lines of:

1. Add a field to the container status with seccomp profile hash (maybe SHA256)
2. When the container (or pod?) is first started, grab the hash of the current profile(s) file(s), and copy the profile file somewhere else, indexed by hash (unless it already exists). Also, verify the hash after copy to avoid TOCTOU bugs
3. When (re)starting a container with a seccomp profile hash already set, use the copied file indexed by the hash rather than the file defined by the spec.
4. Use reference counting to clean up copied files for old versions.

Then again, maybe it's better to just address this at the control-plane level in the seccomp operator to prevent this situation in the first place.

Reply:
I concur with that assessment and have removed the proposed changes in behaviour.

I also agree that the seccomp operator will be a good place to evolve and experiment in this area. Certainly using some of those ideas. :+1: